### PR TITLE
Click a note link will now select the right folder note and not /home

### PR DESCRIPTION
**Pull request by dredav**
## Description
After following a link to an other note the folder /home (All Notes) will be selected. The expected behavior is that the storage folder of the requested note will be selected.

## Screenshot before
Kapture 2019-03-22 at 7 30 09

## Screenshot after
Kapture 2019-03-22 at 7 32 14

## Type of changes

- :large_blue_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :large_blue_circle: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :white_circle: All existing tests have been passed
- :large_blue_circle: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

**Comment by dredav** [comment]
This pull request will also fix the issues #2823 and #2909

File: browser/main/NoteList/index.js [hunk 1]
```diff
@@ -259,13 +259,23 @@ class NoteList extends React.Component {
...
     // first argument event isn't used.
     if (this.notes === null || this.notes.length === 0) {
       return
     }
 
     const selectedNoteKeys = [noteHash]
-    this.focusNote(selectedNoteKeys, noteHash, '/home')
+
+    let locationToSelect = '/home'
+    const notesByHash = data.noteMap.map((note) => note).filter((note) => note.key === noteHash)
```
Comment (ZeroX-DG) [inline browser/main/NoteList/index.js]: Why don't you use `find` instead of `filter`? Can you explain it to me?

Comment (dredav) [reply browser/main/NoteList/index.js]: @ZeroX-DG Thanks for your Feedback. I was inspired by the `handleNoteClick` function in `NoteList/index.js`. I've changed the code and using the `find` function to get the first note and not all notes that matching the key (which always will be one).

File: browser/main/NoteList/index.js [hunk 2]
```diff
@@ -259,13 +259,22 @@ class NoteList extends React.Component {
...
     // first argument event isn't used.
     if (this.notes === null || this.notes.length === 0) {
       return
     }
 
     const selectedNoteKeys = [noteHash]
-    this.focusNote(selectedNoteKeys, noteHash, '/home')
+
+    let locationToSelect = '/home'
+    const noteByHash = data.noteMap.map((note) => note).find(note => { return note.key === noteHash })
```
Comment (ZeroX-DG) [inline browser/main/NoteList/index.js]:
```suggestion
    const noteByHash = data.noteMap.map((note) => note).find(note => note.key === noteHash)
```

**Comment by dredav** [comment]
> I'm sorry but I just can't hold it. Sometimes, I think I have OCD 😄

Thanks for you suggestion. I have applied it.

**Comment by dredav** [comment]
> I'm sorry but I just can't hold it. Sometimes, I think I have OCD 😄

@ZeroX-DG Thanks for you suggestion. I have applied it.

**Pull request merged**